### PR TITLE
Ctda 1052

### PR DESCRIPTION
**Pull request by RobertLaverick**

### CODE REVIEW

File: app/services/RoutingService.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      case None => Left(InvalidMessageCode(s"Invalid Message Type: ${xml.head.label}"))
+      case Some(messageType) =>
+        val officeEither: Either[ParseError, Office] = if(MessageType.arrivalValues.contains(messageType)) {
+          officeOfDestination(xml)
+        }
+        else {
+          officeOfDeparture(xml)
+        }
+
+        officeEither match {
```
Comment (crystalmark) [inline app/services/RoutingService.scala:45]: It's worth checking with Dillon but I think we have to pass the message on to NCTS GB if the office of destination/departure is empty.

**Review by crystalmark — CHANGES_REQUESTED** [review]
Coverage is a little low:
[info] Statement coverage.: 86.84%
[info] Branch coverage....: 100.00%
[info] Coverage reports completed
[error] Coverage is below minimum [86.84% < 90.0%]

File: app/models/MessageTypes.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+trait MessageType extends IeMetadata {
+  def code: String
+  def rootNode: String
+}
+
+sealed trait Directable extends MessageType
+trait ArrivalMessage extends Directable
+trait DepartureMessage extends Directable
+
+object MessageType extends Enumerable.Implicits {
```
[the remark this answers is not in the document]
Comment (RobertLaverick) [reply app/models/MessageTypes.scala:28]: Yeah, I somehow have the exactly wrong messages. Whoops

File: app/services/RoutingService.scala (hunk 1 above)
Comment (RobertLaverick) [reply app/services/RoutingService.scala]: I've checked, and if the departure is empty/missing we're returning an error message.

**Review by crystalmark — APPROVED** [review]

**Pull request merged**